### PR TITLE
ARROW-18426: [Website] Update committer and PMC list

### PR DESCRIPTION
**Pull request by bkmgit**
Affiliation for Ben Baumgold is missing

### CODE REVIEW

**Comment by bkmgit** [comment]
@baumgold Is there an affiliation you would like to have on the Apache Arrow committers website?

File: _data/committers.yml [hunk 1]
```diff
@@ -131,6 +135,10 @@
   role: PMC
   alias: nevime
   affiliation: commutenav
+- name: Nic Crane
```
Comment (alamb) [inline _data/committers.yml:138]: mailing list ref https://lists.apache.org/thread/gr2gkvvlx9xl145on32xp4mff5o8hqf0

File: _data/committers.yml [hunk 2]
```diff
@@ -67,6 +67,10 @@
   role: PMC
   alias: hg
   affiliation: MZ
+- name: Jacob Quinn
```
Comment (alamb) [inline _data/committers.yml:70]: ref https://lists.apache.org/thread/3fyckbljtn38nc54cm75d8owoxkm1p3r

**Review by alamb — APPROVED** [review]
Thank you @bkmgit 

Figuring out how to automate sync from https://home.apache.org/committer-index.html would indeed be wonderful -- maybe we could start with some script that does so that could be run manually,  and then  moving on to running that script automatically

**Comment by baumgold** [comment]
> @baumgold Is there an affiliation you would like to have on the Apache Arrow committers website?

No, thanks.  I'm unaffiliated.

File: _data/committers.yml [hunk 3]
```diff
@@ -204,6 +212,14 @@
   role: Committer
   alias: amolina
   affiliation: Voltron Data
+- name: Ben Baumgold
+  role: Committer
+  alias: baumgold
+  affiliation: ??
```
Comment (bkmgit) [inline _data/committers.yml:218]:
```suggestion
  affiliation: Independent
```

**Comment by alamb** [comment]
Thanks again @bkmgit

**Comment by bkmgit** [comment]
@alamb Welcome. Will start on script soon.

**Pull request merged**